### DISCUSSION TITLE
Latest-ARM32v6 (1.25.0) on Docker doesn't start

### DISCUSSION THREAD

**Discussion by rsoftnl** [dani-garcia/vaultwarden#2509] [Q&A] [answered]
I have Vaultwarden running om my RPi4-B Debian Bullseye within Docker (vaultwarden/server:latest-arm32v6), but after an automatic update to 1.25.0 it will not start. The error message I see in the logs in Portainer is:

/start.sh: 25: exec: /vaultwarden: not found

I had the same issue on Debian Buster before the upgrade to Debian Bullseye, so the upgrade is not the cause of the issue in my opinion. I've started the docker container with 'sh' as entrypoint to be able to debug locally, and running an 'ls -al' reveals that /vaultwarden is available, owned by root and executable... Using /vaultwarden as entrypoint doesn't help. I even removed the "latest" image and have Portainer redownload it, no luck. I'm stumped. Couldn't find a solution on Google, does anybody have an idea where to look? Or am I just being stupid and overlooking something very obvious?

Reverting back to vaultwarden/server:1.24.0-arm32v6 fixes everything. The support string for the 1.24.0 is:

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.24.0
* Web-vault version: v2.25.1
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***********.****.*****.**",
  "domain_origin": "*****://***********.****.*****.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": "*****.**",
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "******************@*****.**",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": false,
  "signups_domains_whitelist": "*****.**",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "LOGIN",
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*************@*****.**",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****.*********.**",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "*************@*****.**",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": 30,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Comment by BlackDex** [comment]
While that should not happen, i do wonder why you would run a 32bit version on a 64bit cpu? There is an arm64 image which better fits your architecture.

**Reply by rsoftnl** [reply]
Thanks, don't recall why I switched from latest to latest-32v6 in the past. I think it had to do with an issue I had at the time, although I can't recall which one. I switched back to 1.24.0 (without arm32v6) and then upgraded to latest. This did the trick. Thanks for pointing me in the right direction.

**Comment by BlackDex** [comment, marked as answer]
Ah, wait that is an issue we already fixed, but not yet released a new version because there are a few other regressions we need to check.
If you use `-testing` instead of `-latest` that should solve it for now.

**Reply by rsoftnl** [reply]
See above, thanks again.
